aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AppSyncCustomNetworkInvoker.java

```diff
@@ -120,11 +121,15 @@ public void run() {
                                    if (!persistentOfflineMutationObject.bucket.equals("")) {
                                        //Fail if S3 Object Manager has not been provided
                                        if (s3ObjectManager == null) {
-                                           persistentMutationsCallback.onFailure(new PersistentMutationsError(persistentOfflineMutationObject.responseClassName,
-                                                   persistentOfflineMutationObject.recordIdentifier, new ApolloNetworkException("S3 upload failed.", new IllegalArgumentException("S3ObjectManager not provided."))));
+                                           //Invoke callback
+                                           if (persistentMutationsCallback != null ) {
+                                               persistentMutationsCallback.onFailure(new PersistentMutationsError(persistentOfflineMutationObject.responseClassName,
+                                                       persistentOfflineMutationObject.recordIdentifier, new ApolloNetworkException("S3 upload failed.", new IllegalArgumentException("S3ObjectManager not provided."))));
+                                           }
+                                           //Remove mutation from queue and mark state as completed.
+                                           setMutationAsCompleted(persistentOfflineMutationObject.recordIdentifier);
```

Comment:
Consider renaming setMutationAsCompleted to indicate that this means a mutation execution is completed.

Reply:
Done.